src/librustc_codegen_llvm/intrinsic.rs

```diff
@@ -1039,6 +1039,29 @@ fn generic_simd_intrinsic(
     llret_ty: &'ll Type,
     span: Span
 ) -> Result<&'ll Value, ()> {
+    // Given a SIMD vector type `x` return the element type and the number of
+    // elements in the vector.
+    fn simd_type_size(bx: &Builder<'a, 'll, 'tcx>, x: Ty<'tcx>) -> (Ty<'tcx>, usize) {
+        let ty = if let ty::Adt(def, substs) = x.sty {
+            assert!(def.repr.simd());
+            let field0_ty = def.non_enum_variant().fields[0].ty(bx.tcx(), substs);
```

Comment:
You could also do this with `layout.field(bx, 0).ty`, which seems cleaner.

Reply:
Where does `layout` come from ?

Reply:
It's what you get from `layout_of(x)` below.